packages/flutter/lib/src/material/switch.dart

```diff
@@ -362,7 +373,8 @@ class _RenderSwitch extends RenderToggleable {
     _drag = HorizontalDragGestureRecognizer()
       ..onStart = _handleDragStart
       ..onUpdate = _handleDragUpdate
-      ..onEnd = _handleDragEnd;
+      ..onEnd = _handleDragEnd
+      ..dragStartBehavior = dragStartBehavior;
```

Comment:
this needs to handle the behavior changing

Reply:
So does that mean I need to have a function that calls `setState` if you update _drag?

Reply:
other way around. Right now your render object widget does nothing when the Switch's dragStartBehavior changes.